sklearn/metrics/_pairwise_distances_reduction.pyx

```diff
@@ -1028,29 +1107,28 @@ cdef class FastEuclideanPairwiseDistancesArgKmin(PairwiseDistancesArgKmin):
         # X and Y are checked by the DatasetsPair implemented as a DenseDenseDatasetsPair
         cdef:
             DenseDenseDatasetsPair datasets_pair = <DenseDenseDatasetsPair> self.datasets_pair
-        self.X, self.Y = datasets_pair.X, datasets_pair.Y
+            ITYPE_t dist_middle_terms_chunks_size = self.Y_n_samples_chunk * self.X_n_samples_chunk
+
+        self.compo = GEMMComponent(
```

Comment:
`compo` is not very explicit, maybe the following would be more explicit?

```suggestion
        self.gemm_term_computer = GEMMComponent(
```

or `euclidean_middle_term_computer` but this is a bit long.

Anyway if the attribute is renamed, the class name should probably be renamed accordingly, e.g. `GEMMTermComputer`.